perfkitbenchmarker/linux_benchmarks/cloudsuite_web_search_benchmark.py

```diff
@@ -0,0 +1,195 @@
+                  'Distribution of query terms. '
+                  'Random and Ziphian distributions are available. '
+                  'Default: Random.')
+flags.DEFINE_integer('cloudsuite_web_search_num_clients',
+                     1,
+                     'Number of client machines.',
+                     lower_bound=1)
+flags.DEFINE_integer('cloudsuite_web_search_ramp_up',
+                     90,
+                     'Benchmark ramp up time in seconds.',
+                     lower_bound=1)
+flags.DEFINE_integer('cloudsuite_web_search_ramp_down',
+                     60,
+                     'Benchmark ramp down time in seconds.',
+                     lower_bound=1)
+flags.DEFINE_integer('cloudsuite_web_search_steady_state',
+                     60,
+                     'Benchmark steady state time in seconds.',
+                     lower_bound=1)
+flags.DEFINE_integer('cloudsuite_web_search_scale',
+                     50,
+                     'Number of simulated web search users.',
+                     lower_bound=1)
+
+BENCHMARK_NAME = 'cloudsuite_web_search'
+BENCHMARK_CONFIG = """
+cloudsuite_web_search:
+  description: >
+    Run Cloudsuite Web Search benchmark. Specify the number of
+    clients with --num_vms.
+  vm_groups:
+    servers:
+      vm_spec: *default_single_core
+      disk_spec: *default_500_gb
+    clients:
+      vm_spec: *default_single_core
+      vm_count: 1
+"""
+
+DISK_PATH = '/scratch'
+
+
+def GetConfig(user_config):
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  if FLAGS['num_vms'].present:
+    config['vm_groups']['clients']['vm_count'] = FLAGS.num_vms
+  return config
+
+
+def Prepare(benchmark_spec):
+  """Install docker. Pull the required images from DockerHub.
+
+  Start Solr index node and client.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  servers = benchmark_spec.vm_groups['servers'][0]
+  clients = benchmark_spec.vm_groups['clients']
+
+  def PrepareCommon(vm):
+    if not docker.IsInstalled(vm):
+      vm.Install('docker')
+
+  def PrepareServer(vm):
+    PrepareCommon(vm)
+    server_cmd = ('sudo echo \'DOCKER_OPTS="-g %s"\''
+                  '| sudo tee /etc/default/docker > /dev/null' % (DISK_PATH))
+    stdout, _ = vm.RemoteCommand(server_cmd, should_log=True)
+
+    server_cmd = 'sudo service docker restart'
+    stdout, _ = vm.RemoteCommand(server_cmd, should_log=True)
+
+    vm.RemoteCommand('sudo docker pull cloudsuite/web-search:server')
+
+    server_cmd = ('sudo docker run -d --net host '
+                  '--name server cloudsuite/web-search:server %s 1' %
+                  (FLAGS.cloudsuite_web_search_server_heap_size))
+
+    stdout, _ = servers.RemoteCommand(server_cmd, should_log=True)
+
+  def PrepareClient(vm):
+    PrepareCommon(vm)
+    vm.RemoteCommand('sudo docker pull cloudsuite/web-search:client')
+    time.sleep(120)
```

Comment:
Please add a comment explaining what the sleep is for, or wait/poll on the server.